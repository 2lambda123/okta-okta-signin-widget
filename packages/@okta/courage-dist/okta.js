/*! THIS FILE IS GENERATED FROM PACKAGE @okta/courage@4.6.0-alpha.3660.g8faa6ac */
module.exports =
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 0);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

module.exports = __webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

throw new Error("Module build failed: Error: Cannot find module '@okta/babel-plugin-handlebars-inline-precompile' from '/Users/anipendakur/okta/okta-signin-widget/packages/@okta/courage-for-signin-widget'\n    at Function.module.exports [as sync] (/Users/anipendakur/okta/okta-signin-widget/node_modules/resolve/lib/sync.js:71:15)\n    at resolveStandardizedName (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/files/plugins.js:101:31)\n    at resolvePlugin (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/files/plugins.js:54:10)\n    at loadPlugin (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/files/plugins.js:62:20)\n    at createDescriptor (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)\n    at /Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:109:50\n    at Array.map (<anonymous>)\n    at createDescriptors (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)\n    at createPluginDescriptors (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:105:10)\n    at /Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:63:53\n    at cachedFunction (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/caching.js:62:27)\n    at cachedFunction.next (<anonymous>)\n    at evaluateSync (/Users/anipendakur/okta/okta-signin-widget/node_modules/gensync/index.js:244:28)\n    at sync (/Users/anipendakur/okta/okta-signin-widget/node_modules/gensync/index.js:84:14)\n    at plugins (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-descriptors.js:28:77)\n    at mergeChainOpts (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-chain.js:319:26)\n    at /Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-chain.js:283:7\n    at Generator.next (<anonymous>)\n    at buildRootChain (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/config-chain.js:68:36)\n    at buildRootChain.next (<anonymous>)\n    at loadPrivatePartialConfig (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/partial.js:95:62)\n    at loadPrivatePartialConfig.next (<anonymous>)\n    at Function.<anonymous> (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/partial.js:120:25)\n    at Generator.next (<anonymous>)\n    at evaluateSync (/Users/anipendakur/okta/okta-signin-widget/node_modules/gensync/index.js:244:28)\n    at Function.sync (/Users/anipendakur/okta/okta-signin-widget/node_modules/gensync/index.js:84:14)\n    at Object.<anonymous> (/Users/anipendakur/okta/okta-signin-widget/node_modules/@babel/core/lib/config/index.js:43:61)\n    at Object.<anonymous> (/Users/anipendakur/okta/okta-signin-widget/node_modules/babel-loader/lib/index.js:151:26)\n    at Generator.next (<anonymous>)\n    at asyncGeneratorStep (/Users/anipendakur/okta/okta-signin-widget/node_modules/babel-loader/lib/index.js:3:103)");

/***/ })
/******/ ]);
//# sourceMappingURL=okta.js.map