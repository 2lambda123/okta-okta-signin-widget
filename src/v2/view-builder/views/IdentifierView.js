import { View, loc, createButton } from 'okta';
import BaseView from '../internals/BaseView';
import BaseForm from '../internals/BaseForm';
import BaseFooter from '../internals/BaseFooter';
import Util from '../../../util/Util';

const Body = BaseForm.extend({

  title: loc('primaryauth.title'),
  save: loc('oform.next', 'login'),
  initialize () {
    BaseForm.prototype.initialize.apply(this, arguments);
    if (this.options.appState.hasRemediationForm('launch-authenticator')) {
      this.add(View.extend({
        className: 'sign-in-with-device-option',
        template: `
          <div class="okta-verify-container"></div>
          <div class="separation-line"><span>OR</span></div>
        `,
        initialize () {
          const appState = this.options.appState;
          this.listenTo(this.options.appState, 'launch-universal-link', () => {
            const deviceChallenge = this.options.appState.get(
              this.options.appState.getCurrentViewState().relatesTo
            );
            Util.redirect(deviceChallenge.href); 
            // Util.redirect('https://z9pm.adj.st/my/path?adjust_t=klm012&adjust_campaign=Getaway_Vacation_Campaign_2-0&adjust_adgroup=35-42&adjust_creative=island_sunset');
          });
          this.add(createButton({
            className: 'button-secondary',
            title: 'Sign in using Okta Verify',
            click () { 
              appState.trigger('invokeAction', 'launch-authenticator');
            }
          }), '.okta-verify-container');
        }
      }), '.o-form-fieldset-container', false, true);
    }
  }
});

const Footer = BaseFooter.extend({
  links () {
    const baseUrl = this.options.settings.get('baseUrl');
    let href = baseUrl + '/help/login';
    if (this.options.settings.get('helpLinks.help') ) {
      href = this.options.settings.get('helpLinks.help');
    }
    const signupLinkObj = {
      'type': 'link',
      'label': 'Sign up',
      'name': 'enroll',
      'actionPath': 'select-enroll-profile',
    };
    const links = [
      {
        'name': 'help',
        'label': 'Need help signing in?',
        'href': href,
      },
    ];
    if (this.options.appState.hasRemediationForm('select-enroll-profile')) {
      links.push(signupLinkObj);
    }
    return links;
  }
});

export default BaseView.extend({
  Body,
  Footer,
});
