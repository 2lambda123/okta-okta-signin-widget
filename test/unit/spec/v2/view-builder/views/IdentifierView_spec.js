import { Model } from 'okta';
import IdentifierView from 'v2/view-builder/views/IdentifierView';
import AppState from 'v2/models/AppState';
import Settings from 'models/Settings';
import XHRIdentifyWithPassword
  from '../../../../../../playground/mocks/data/idp/idx/identify-with-password.json';
import XHRIdentifyWithThirdPartyIdps
  from '../../../../../../playground/mocks/data/idp/idx/identify-with-third-party-idps.json';

describe('v2/view-builder/views/CaptchaView', function() {
  let testContext;
  beforeEach(function() { 
    testContext = {};
    testContext.init = (remediations = XHRIdentifyWithThirdPartyIdps.remediation.value) => {
      const appState = new AppState();
      appState.set('remediations', remediations);
      const settings = new Settings({ baseUrl: 'http://localhost:3000' });
      testContext.view = new IdentifierView({
        appState,
        settings,
        currentViewState: {},
        model: new Model(),
      });
      testContext.view.render();
    };
  });
  afterEach(function() {
    jest.resetAllMocks();
  });

  it('view renders forgot password link correctly with mutiple IDPs', function() {
    jest.spyOn(AppState.prototype, 'hasRemediationObject').mockReturnValue(true);
    jest.spyOn(AppState.prototype, 'getActionByPath').mockReturnValue(true);
    jest.spyOn(AppState.prototype, 'isIdentifierOnlyView').mockReturnValue(false);
    testContext.init();

    // The forgot password link should NOT be in the siw-main-footer
    expect(testContext.view.el).toMatchSnapshot('With Multiple IDPs');
  });

  it('view renders forgot password link correctly with no IDPs', function() {
    jest.spyOn(AppState.prototype, 'hasRemediationObject').mockReturnValue(true);
    jest.spyOn(AppState.prototype, 'getActionByPath').mockReturnValue(true);
    jest.spyOn(AppState.prototype, 'isIdentifierOnlyView').mockReturnValue(false);
    testContext.init(XHRIdentifyWithPassword.remediation.value);
    
    // The forgot password link should be in the siw-main-footer
    expect(testContext.view.el).toMatchSnapshot('With Multiple IDPs');
  });
});
