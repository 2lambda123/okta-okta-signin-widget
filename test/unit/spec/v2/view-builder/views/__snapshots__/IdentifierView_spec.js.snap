// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v2/view-builder/views/CaptchaView view renders forgot password link correctly with mutiple IDPs: With Multiple IDPs 1`] = `
<div
  class="siw-main-view "
>
  <div
    class="siw-main-header"
  >
    <div />
  </div>
  <div
    class="siw-main-body"
  >
    <form
      action="/"
      class="ion-form o-form o-form-edit-mode"
      data-se="o-form"
      id="form1"
      method="POST"
      slot="content"
    >
      <div
        class="o-form-content o-form-theme clearfix"
        data-se="o-form-content"
      >
        <h2
          class="okta-form-title o-form-head"
          data-se="o-form-head"
        >
          Sign In
        </h2>
        <div
          class="o-form-error-container"
          data-se="o-form-error-container"
        />
        <div
          class="o-form-fieldset-container"
          data-se="o-form-fieldset-container"
        >
          <div
            class="sign-in-with-device-option"
          >
            <div
              class="okta-verify-container"
            >
              <a
                class="button link-button link-button-icon"
                data-se="button"
                href="#"
              >
                <span
                  class="icon okta-verify-authenticator"
                />
                Sign in using Okta Verify on this device
              </a>
            </div>
            <div
              class="separation-line"
            >
              <span>
                or
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="o-form-button-bar"
      >
        <input
          class="button button-primary"
          data-type="save"
          type="submit"
          value="Next"
        />
        <div
          class="links-container"
        >
          <a
            class="link js-forgot-password"
            data-se="forgot-password"
            href="#"
          >
            Forgot password?
          </a>
        </div>
        <div
          class="sign-in-with-idp"
        >
          <div
            class="separation-line"
          >
            <span>
              OR
            </span>
          </div>
          <div
            class="okta-idps-container"
          >
            <a
              class="social-auth-button social-auth-google-button link-button"
              data-se="social-auth-google-button"
              href="http://localhost:3000/sso/idps/google-123?stateToken=eyJ6aXAiOiJERUYiLCJhbGlhcyI6ImVuY3J5cHRpb25rZXkiLCJ2ZXIiOiIxIiwib2lkIjoiMDBvczI0VHZiWHlqOVFLSm4wZzMiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIn0..aDi6JnIZwCRzk7RN.xQ1R9jj43bV5S17F1juzgTZGd9Gq7VvuE4Hc1V_tMVcTWiVQ1Ntim1tQPZtmnpfOifu-N_quGGDSdgiidb_2xU1m9SbDMJhV-Yp6E07tABwjaRd_ekuRzDFw7Jxknz33LtSwQOTQ-WiH2o6JikLiEz2BQbgoizIc2izbtdod7HJg0HVnMZ9ZlBqyBG1-EISI_xPypq8uzP8n79hP_Zz41RI7VJA35yfTuNLOX_k6B-mfeVKf4HyFsKm62XWQgcPIxhjxBMqmyZow2toNclV3sIqgw7I5tKCLQSmSnKbFxxuT4-G31BdaVOALAe9Z89zlluTYaKAPOr86RMsqaGKnQFaplc_0PiPbreKhVgvSeeJgqX2RwnLgckLLiRo7TRDs2kLhGY2ope0AeA9TSsTVdJzsScftZWKgh9iHpXjS-kGcbRx0etu4iTtbHOu3rDIfIcvvt55mfvA66wzy1CCxHt4WYNnBKHX0fIOW_fa_-RYGYug9YRV5G6nQ6V-CfHoxmEsMhsoFJu0hei34_SJv15w2l3vxxBytrWSWi5qUfm5zGjNlx8e9n1Sf_eAqXCfLhBLK4_14jwtjNbWOZCdg5dwzxQiQWDItBjijEjdQrK0i6tw2Rp-IMJD1-4_ZfFZDmAXgZZtBYc3kdmumgYpKeYUJJgw0ZJWoG-Xr0bbzGGMx46yHzMpDbSTpiWhKGytQPbNja8sf_eeOKx_AAosamDUub9yuZJb0-Nj0xvXZ89J0m_09wa2Z3G-zY01sv9ONkXMFzRVwAb2bHmGle082bq33-7Klk7_ZzzkBROJhgDHQcw5QibGWaqYqscgKv2NQV8ebGJO_BHU46p1T3MQzStxRZ2EZua9qQwsmL8P5yboNDt2YmYnUvaOcGfeAqwgovqNDQ0H4u-D5psFmiU1STLOlN5pSAauKe4VxlLxphiirrmiNOOOW0XTwaQ1vtPz8gFlXsmGB-0zcsySG6A6HJ49eOeEI0J2REy2dlFRxzdKthANM2xFc_AIgas9mcNhSWtmVEtMxv7N0xqGAJbxaJC6U4kDDXdImZVaovz4lgRFkIh3aUXgUMX558u9MBeF6Q7z3piIpT6A4I1ww_eDNM02Vew.inRUXNhsc6Evt7GAb8DPAA"
            >
              Sign in with Google
            </a>
            <a
              class="social-auth-button social-auth-facebook-button link-button"
              data-se="social-auth-facebook-button"
              href="http://localhost:3000/sso/idps/facebook-123?stateToken=eyJ6aXAiOiJERUYiLCJhbGlhcyI6ImVuY3J5cHRpb25rZXkiLCJ2ZXIiOiIxIiwib2lkIjoiMDBvczI0VHZiWHlqOVFLSm4wZzMiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIn0..aDi6JnIZwCRzk7RN.xQ1R9jj43bV5S17F1juzgTZGd9Gq7VvuE4Hc1V_tMVcTWiVQ1Ntim1tQPZtmnpfOifu-N_quGGDSdgiidb_2xU1m9SbDMJhV-Yp6E07tABwjaRd_ekuRzDFw7Jxknz33LtSwQOTQ-WiH2o6JikLiEz2BQbgoizIc2izbtdod7HJg0HVnMZ9ZlBqyBG1-EISI_xPypq8uzP8n79hP_Zz41RI7VJA35yfTuNLOX_k6B-mfeVKf4HyFsKm62XWQgcPIxhjxBMqmyZow2toNclV3sIqgw7I5tKCLQSmSnKbFxxuT4-G31BdaVOALAe9Z89zlluTYaKAPOr86RMsqaGKnQFaplc_0PiPbreKhVgvSeeJgqX2RwnLgckLLiRo7TRDs2kLhGY2ope0AeA9TSsTVdJzsScftZWKgh9iHpXjS-kGcbRx0etu4iTtbHOu3rDIfIcvvt55mfvA66wzy1CCxHt4WYNnBKHX0fIOW_fa_-RYGYug9YRV5G6nQ6V-CfHoxmEsMhsoFJu0hei34_SJv15w2l3vxxBytrWSWi5qUfm5zGjNlx8e9n1Sf_eAqXCfLhBLK4_14jwtjNbWOZCdg5dwzxQiQWDItBjijEjdQrK0i6tw2Rp-IMJD1-4_ZfFZDmAXgZZtBYc3kdmumgYpKeYUJJgw0ZJWoG-Xr0bbzGGMx46yHzMpDbSTpiWhKGytQPbNja8sf_eeOKx_AAosamDUub9yuZJb0-Nj0xvXZ89J0m_09wa2Z3G-zY01sv9ONkXMFzRVwAb2bHmGle082bq33-7Klk7_ZzzkBROJhgDHQcw5QibGWaqYqscgKv2NQV8ebGJO_BHU46p1T3MQzStxRZ2EZua9qQwsmL8P5yboNDt2YmYnUvaOcGfeAqwgovqNDQ0H4u-D5psFmiU1STLOlN5pSAauKe4VxlLxphiirrmiNOOOW0XTwaQ1vtPz8gFlXsmGB-0zcsySG6A6HJ49eOeEI0J2REy2dlFRxzdKthANM2xFc_AIgas9mcNhSWtmVEtMxv7N0xqGAJbxaJC6U4kDDXdImZVaovz4lgRFkIh3aUXgUMX558u9MBeF6Q7z3piIpT6A4I1ww_eDNM02Vew.inRUXNhsc6Evt7GAb8DPAA"
            >
              Sign in with Facebook
            </a>
            <a
              class="social-auth-button social-auth-linkedin-button link-button"
              data-se="social-auth-linkedin-button"
              href="http://localhost:3000/sso/idps/linkedin-123?stateToken=eyJ6aXAiOiJERUYiLCJhbGlhcyI6ImVuY3J5cHRpb25rZXkiLCJ2ZXIiOiIxIiwib2lkIjoiMDBvczI0VHZiWHlqOVFLSm4wZzMiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIn0..aDi6JnIZwCRzk7RN.xQ1R9jj43bV5S17F1juzgTZGd9Gq7VvuE4Hc1V_tMVcTWiVQ1Ntim1tQPZtmnpfOifu-N_quGGDSdgiidb_2xU1m9SbDMJhV-Yp6E07tABwjaRd_ekuRzDFw7Jxknz33LtSwQOTQ-WiH2o6JikLiEz2BQbgoizIc2izbtdod7HJg0HVnMZ9ZlBqyBG1-EISI_xPypq8uzP8n79hP_Zz41RI7VJA35yfTuNLOX_k6B-mfeVKf4HyFsKm62XWQgcPIxhjxBMqmyZow2toNclV3sIqgw7I5tKCLQSmSnKbFxxuT4-G31BdaVOALAe9Z89zlluTYaKAPOr86RMsqaGKnQFaplc_0PiPbreKhVgvSeeJgqX2RwnLgckLLiRo7TRDs2kLhGY2ope0AeA9TSsTVdJzsScftZWKgh9iHpXjS-kGcbRx0etu4iTtbHOu3rDIfIcvvt55mfvA66wzy1CCxHt4WYNnBKHX0fIOW_fa_-RYGYug9YRV5G6nQ6V-CfHoxmEsMhsoFJu0hei34_SJv15w2l3vxxBytrWSWi5qUfm5zGjNlx8e9n1Sf_eAqXCfLhBLK4_14jwtjNbWOZCdg5dwzxQiQWDItBjijEjdQrK0i6tw2Rp-IMJD1-4_ZfFZDmAXgZZtBYc3kdmumgYpKeYUJJgw0ZJWoG-Xr0bbzGGMx46yHzMpDbSTpiWhKGytQPbNja8sf_eeOKx_AAosamDUub9yuZJb0-Nj0xvXZ89J0m_09wa2Z3G-zY01sv9ONkXMFzRVwAb2bHmGle082bq33-7Klk7_ZzzkBROJhgDHQcw5QibGWaqYqscgKv2NQV8ebGJO_BHU46p1T3MQzStxRZ2EZua9qQwsmL8P5yboNDt2YmYnUvaOcGfeAqwgovqNDQ0H4u-D5psFmiU1STLOlN5pSAauKe4VxlLxphiirrmiNOOOW0XTwaQ1vtPz8gFlXsmGB-0zcsySG6A6HJ49eOeEI0J2REy2dlFRxzdKthANM2xFc_AIgas9mcNhSWtmVEtMxv7N0xqGAJbxaJC6U4kDDXdImZVaovz4lgRFkIh3aUXgUMX558u9MBeF6Q7z3piIpT6A4I1ww_eDNM02Vew.inRUXNhsc6Evt7GAb8DPAA"
            >
              Sign in with LinkedIn
            </a>
            <a
              class="social-auth-button social-auth-microsoft-button link-button"
              data-se="social-auth-microsoft-button"
              href="http://localhost:3000/sso/idps/microsoft-123?stateToken=eyJ6aXAiOiJERUYiLCJhbGlhcyI6ImVuY3J5cHRpb25rZXkiLCJ2ZXIiOiIxIiwib2lkIjoiMDBvczI0VHZiWHlqOVFLSm4wZzMiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIn0..aDi6JnIZwCRzk7RN.xQ1R9jj43bV5S17F1juzgTZGd9Gq7VvuE4Hc1V_tMVcTWiVQ1Ntim1tQPZtmnpfOifu-N_quGGDSdgiidb_2xU1m9SbDMJhV-Yp6E07tABwjaRd_ekuRzDFw7Jxknz33LtSwQOTQ-WiH2o6JikLiEz2BQbgoizIc2izbtdod7HJg0HVnMZ9ZlBqyBG1-EISI_xPypq8uzP8n79hP_Zz41RI7VJA35yfTuNLOX_k6B-mfeVKf4HyFsKm62XWQgcPIxhjxBMqmyZow2toNclV3sIqgw7I5tKCLQSmSnKbFxxuT4-G31BdaVOALAe9Z89zlluTYaKAPOr86RMsqaGKnQFaplc_0PiPbreKhVgvSeeJgqX2RwnLgckLLiRo7TRDs2kLhGY2ope0AeA9TSsTVdJzsScftZWKgh9iHpXjS-kGcbRx0etu4iTtbHOu3rDIfIcvvt55mfvA66wzy1CCxHt4WYNnBKHX0fIOW_fa_-RYGYug9YRV5G6nQ6V-CfHoxmEsMhsoFJu0hei34_SJv15w2l3vxxBytrWSWi5qUfm5zGjNlx8e9n1Sf_eAqXCfLhBLK4_14jwtjNbWOZCdg5dwzxQiQWDItBjijEjdQrK0i6tw2Rp-IMJD1-4_ZfFZDmAXgZZtBYc3kdmumgYpKeYUJJgw0ZJWoG-Xr0bbzGGMx46yHzMpDbSTpiWhKGytQPbNja8sf_eeOKx_AAosamDUub9yuZJb0-Nj0xvXZ89J0m_09wa2Z3G-zY01sv9ONkXMFzRVwAb2bHmGle082bq33-7Klk7_ZzzkBROJhgDHQcw5QibGWaqYqscgKv2NQV8ebGJO_BHU46p1T3MQzStxRZ2EZua9qQwsmL8P5yboNDt2YmYnUvaOcGfeAqwgovqNDQ0H4u-D5psFmiU1STLOlN5pSAauKe4VxlLxphiirrmiNOOOW0XTwaQ1vtPz8gFlXsmGB-0zcsySG6A6HJ49eOeEI0J2REy2dlFRxzdKthANM2xFc_AIgas9mcNhSWtmVEtMxv7N0xqGAJbxaJC6U4kDDXdImZVaovz4lgRFkIh3aUXgUMX558u9MBeF6Q7z3piIpT6A4I1ww_eDNM02Vew.inRUXNhsc6Evt7GAb8DPAA"
            >
              Sign in with Microsoft
            </a>
            <a
              class="social-auth-button social-auth-apple-button link-button"
              data-se="social-auth-apple-button"
              href="http://localhost:3000/sso/idps/apple-123?stateToken=eyJ6aXAiOiJERUYiLCJhbGlhcyI6ImVuY3J5cHRpb25rZXkiLCJ2ZXIiOiIxIiwib2lkIjoiMDBvczI0VHZiWHlqOVFLSm4wZzMiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiZGlyIn0..aDi6JnIZwCRzk7RN.xQ1R9jj43bV5S17F1juzgTZGd9Gq7VvuE4Hc1V_tMVcTWiVQ1Ntim1tQPZtmnpfOifu-N_quGGDSdgiidb_2xU1m9SbDMJhV-Yp6E07tABwjaRd_ekuRzDFw7Jxknz33LtSwQOTQ-WiH2o6JikLiEz2BQbgoizIc2izbtdod7HJg0HVnMZ9ZlBqyBG1-EISI_xPypq8uzP8n79hP_Zz41RI7VJA35yfTuNLOX_k6B-mfeVKf4HyFsKm62XWQgcPIxhjxBMqmyZow2toNclV3sIqgw7I5tKCLQSmSnKbFxxuT4-G31BdaVOALAe9Z89zlluTYaKAPOr86RMsqaGKnQFaplc_0PiPbreKhVgvSeeJgqX2RwnLgckLLiRo7TRDs2kLhGY2ope0AeA9TSsTVdJzsScftZWKgh9iHpXjS-kGcbRx0etu4iTtbHOu3rDIfIcvvt55mfvA66wzy1CCxHt4WYNnBKHX0fIOW_fa_-RYGYug9YRV5G6nQ6V-CfHoxmEsMhsoFJu0hei34_SJv15w2l3vxxBytrWSWi5qUfm5zGjNlx8e9n1Sf_eAqXCfLhBLK4_14jwtjNbWOZCdg5dwzxQiQWDItBjijEjdQrK0i6tw2Rp-IMJD1-4_ZfFZDmAXgZZtBYc3kdmumgYpKeYUJJgw0ZJWoG-Xr0bbzGGMx46yHzMpDbSTpiWhKGytQPbNja8sf_eeOKx_AAosamDUub9yuZJb0-Nj0xvXZ89J0m_09wa2Z3G-zY01sv9ONkXMFzRVwAb2bHmGle082bq33-7Klk7_ZzzkBROJhgDHQcw5QibGWaqYqscgKv2NQV8ebGJO_BHU46p1T3MQzStxRZ2EZua9qQwsmL8P5yboNDt2YmYnUvaOcGfeAqwgovqNDQ0H4u-D5psFmiU1STLOlN5pSAauKe4VxlLxphiirrmiNOOOW0XTwaQ1vtPz8gFlXsmGB-0zcsySG6A6HJ49eOeEI0J2REy2dlFRxzdKthANM2xFc_AIgas9mcNhSWtmVEtMxv7N0xqGAJbxaJC6U4kDDXdImZVaovz4lgRFkIh3aUXgUMX558u9MBeF6Q7z3piIpT6A4I1ww_eDNM02Vew.inRUXNhsc6Evt7GAb8DPAA"
            >
              Sign in with Apple
            </a>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    class="siw-main-footer"
  >
    <div
      class="auth-footer"
    >
      <a
        class="link js-unlock"
        data-se="unlock"
        href="#"
      >
        Unlock account?
      </a>
      <a
        class="link js-enroll"
        data-se="enroll"
        href="#"
      >
        Sign up
      </a>
      <a
        class="link js-help"
        data-se="help"
        href="http://localhost:3000/help/login"
      >
        Help
      </a>
    </div>
  </div>
</div>
`;

exports[`v2/view-builder/views/CaptchaView view renders forgot password link correctly with no IDPs: With Multiple IDPs 1`] = `
<div
  class="siw-main-view "
>
  <div
    class="siw-main-header"
  >
    <div />
  </div>
  <div
    class="siw-main-body"
  >
    <form
      action="/"
      class="ion-form o-form o-form-edit-mode"
      data-se="o-form"
      id="form2"
      method="POST"
      slot="content"
    >
      <div
        class="o-form-content o-form-theme clearfix"
        data-se="o-form-content"
      >
        <h2
          class="okta-form-title o-form-head"
          data-se="o-form-head"
        >
          Sign In
        </h2>
        <div
          class="o-form-error-container"
          data-se="o-form-error-container"
        />
        <div
          class="o-form-fieldset-container"
          data-se="o-form-fieldset-container"
        >
          <div
            class="sign-in-with-device-option"
          >
            <div
              class="okta-verify-container"
            >
              <a
                class="button link-button link-button-icon"
                data-se="button"
                href="#"
              >
                <span
                  class="icon okta-verify-authenticator"
                />
                Sign in using Okta Verify on this device
              </a>
            </div>
            <div
              class="separation-line"
            >
              <span>
                or
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="o-form-button-bar"
      >
        <input
          class="button button-primary"
          data-type="save"
          type="submit"
          value="Next"
        />
      </div>
    </form>
  </div>
  <div
    class="siw-main-footer"
  >
    <div
      class="auth-footer"
    >
      <a
        class="link js-forgot-password"
        data-se="forgot-password"
        href="#"
      >
        Forgot password?
      </a>
      <a
        class="link js-unlock"
        data-se="unlock"
        href="#"
      >
        Unlock account?
      </a>
      <a
        class="link js-enroll"
        data-se="enroll"
        href="#"
      >
        Sign up
      </a>
      <a
        class="link js-help"
        data-se="help"
        href="http://localhost:3000/help/login"
      >
        Help
      </a>
    </div>
  </div>
</div>
`;
